app/models/ability.rb

```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user
+
+    can :read, :all
+
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, user_id: user.id
+    end
```

Comment:
- [ ] Kudos adding the ability 👍 especially the admin management , however when a user is connected , he should be granted the access to manage all 👍 ,Kindly improve  that 💯 

```ruby
 # app/models/ability.rb

 class Ability
 include CanCan::Ability

     def initialize(user)

         ....

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user 
         end
     end
 end
```